internal/tss/utils/transaction_builder.go

```diff
@@ -0,0 +1,49 @@
+package utils
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+
+	xdr3 "github.com/stellar/go-xdr/xdr3"
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/txnbuild"
+	"github.com/stellar/go/xdr"
+)
+
+func BuildOriginalTransaction(txOpXDRs []string) (*txnbuild.Transaction, error) {
+	var operations []txnbuild.Operation
+	for _, opXDR := range txOpXDRs {
+		decodedBytes, err := base64.StdEncoding.DecodeString(opXDR)
+		if err != nil {
+			return nil, fmt.Errorf("decoding Operation XDR string")
+		}
+		var decodedOp xdr.Operation
+		_, err = xdr3.Unmarshal(bytes.NewReader(decodedBytes), &decodedOp)
+
+		if err != nil {
+			return nil, fmt.Errorf("decoding xdr into xdr Operation: %w", err)
+		}
+		// for now, we assume that all operations are Payment operations
+		paymentOp := txnbuild.Payment{}
+		err = paymentOp.FromXDR(decodedOp)
+		if err != nil {
+			return nil, fmt.Errorf("unmarshaling xdr into Operation: %w", err)
+		}
+		err = paymentOp.Validate()
+		if err != nil {
+			return nil, fmt.Errorf("invalid Operation: %w", err)
+		}
+		operations = append(operations, &paymentOp)
```

Comment:
We could leave this function more easily extendible like so wdyt?
```suggestion
		switch decodedOp.Body.Type {
		case xdr.OperationTypePayment:
			paymentOp := txnbuild.Payment{}
			err = paymentOp.FromXDR(decodedOp)
			if err != nil {
				return nil, fmt.Errorf("unmarshaling xdr into Operation: %w", err)
			}

			err = paymentOp.Validate()
			if err != nil {
				return nil, fmt.Errorf("invalid Operation: %w", err)
			}

			operations = append(operations, &paymentOp)

		default:
			return nil, fmt.Errorf("unimplemented operation type %s", decodedOp.Body.Type)
		}
```